src/core/account/account.ts

```diff
@@ -0,0 +1,139 @@
+import type { AccountAuthenticator } from "../../transactions/authenticator/account";
+import { HexInput, SigningSchemeInput } from "../../types";
+import type { AccountAddress, AccountAddressInput } from "../accountAddress";
+import { Ed25519PrivateKey } from "../crypto";
+import { Ed25519Account } from "./ed25519";
+import { type PrivateKeyInput, SingleKeyAccount } from "./singleKey";
+
+export interface CreateEd25519AccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy?: true;
+}
+
+export interface CreateEd25519SingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+  legacy: false;
+}
+
+export interface CreateSingleKeyAccountFromPrivateKeyArgs {
+  privateKey: Exclude<PrivateKeyInput, Ed25519PrivateKey>;
+  address?: AccountAddressInput;
+  legacy?: false;
+}
+
+export interface CreateAccountFromPrivateKeyArgs {
```

Comment:
This type overload can be confusing especially for devs that are less TS advanced - let's add a brief explanation who each type eventually resolved to `CreateAccountFromPrivateKeyArgs`